packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3234,9 +3234,16 @@ class EditableTextState extends State<EditableText>
   void didChangeDependencies() {
     super.didChangeDependencies();
 
-    _style = MediaQuery.boldTextOf(context)
-        ? widget.style.merge(const TextStyle(fontWeight: FontWeight.bold))
-        : widget.style;
+    // Apply platform settings to text style.
+    final ui.TypographySettings? typographySettings = MediaQuery.maybeTypographySettingsOf(context);
```

Comment:
Previously we only created a new text style if a value changed. Now, we will always create a new text style, even though it will frequently be the same as the widget's original style. Do you think it'd be worth adding logic to reuse the widget's style if it is unchanged?

One way this could be implemented is by introducing a `TypographySettings.empty`. If `MediaQuery.boldTextOf` is `false` and `typographySettings == TypographySettings.empty`, we can reuse `widget.style`. Just a thought, I don't feel strongly about this.